pegasus/sites.v3/code.org/public/learn/index.haml

```diff
@@ -130,6 +130,7 @@ style_min: false
   - tutorial[:string_platforms]                     = hoc_s(prefix + "string_platforms")
   - tutorial[:string_detail_platforms]              = hoc_s(prefix + "string_detail_platforms")
   - tutorial[:string_detail_programming_languages]  = hoc_s(prefix + "string_detail_programming_languages")
+  - tutorial[:teachers_notes]                       = CDO.curriculum_url(request.locale, tutorial[:teachers_notes] ? URI.encode(tutorial[:teachers_notes]) : nil, false)
```

Comment:
why is it necessary to `URI.encode` here?

Reply:
Without encoding, we see this error:

`bad URI(is not URI?): https://www.codehs.com/hoc_guide_karel`

According to https://stackoverflow.com/questions/15700784/how-to-fix-bad-uri-is-not-uri this error has to do with how the browser handles special characters (like in full URIs)

Reply:
makes sense! Let's move this into the `curriculum_url` method itself, then, since it's an issue with calling URI.parse

Reply:
Done!